docs/wiki/blocks/block-body.md

```diff
@@ -0,0 +1,70 @@
+| Feature       | **Plain** (0x00) or **Coinbase** (0x01) | 1 byte              |
+| Commitment    | Pedersen commitment                     | 33 bytes            |
+
+## Outputs
+
+Just as in case of the inputs, the outputs consist of **Feature** and a **Commitment** but in addition they also hold **Range proofs**
+
+| Field         | Description                             | Size                |
+|:--------------|:----------------------------------------|:--------------------|
+| Feature       | **Plain** (0x00) or **Coinbase** (0x01) | 1 byte              |
+| Commitment    | Pedersen commitment                     | 33 bytes            |
+| Range proof   | Proves the transaction is net positive  | 8 bytes             |
+
+## Kernels
+
+Every kernel starts with **Kernel feature** which indicates how it should be deserialized. In each variant, the kernel concludes with an **Excess commitment** and a **Excess signature**. What differs them is what is in the middle. We will describe each variant separately
+
+### Coinbase kernel
+
+The most basic kernel that does not contain **Fee** nor **Lock height**.
+
+| Field             | Description                                                                                       | Size                |
+|:------------------|:--------------------------------------------------------------------------------------------------|:--------------------|
+| Feature           | **Plain** (0x00), **Coinbase** (0x01), **Height Locked** (0x02) or **No recent duplicate** (0x03) | 1 byte              |
+| Excess commitment | Pedersen commitment                                                                               | 33 bytes            |
+| Excess signature  | Schnorr signature                                                                                 | 64 bytes            |
+
+### Plain kernel
+
+A kernel resulting from a transaction, unlike **Coinbase** kernel it also includes **Fee** billed for the transaction.
+
+| Field             | Description                                                                                       | Size                |
+|:------------------|:--------------------------------------------------------------------------------------------------|:--------------------|
+| Feature           | **Plain** (0x00), **Coinbase** (0x01), **Height Locked** (0x02) or **No recent duplicate** (0x03) | 1 byte              |
+| Fee               | Transaction fee                                                                                   | 8 bytes             |
+| Excess commitment | Pedersen commitment                                                                               | 33 bytes            |
+| Excess signature  | Schnorr signature                                                                                 | 64 bytes            |
+
+### Height locked / no recent duplicate / V1 protocol kernel
```

Comment:
Height locked and NRD kernels should each have their own entry, as their fields have different meanings.

Reply:
I split them into separate sections, could you help me with the content? I don't really understand the difference between them.

Reply:
For backgroud, here's the mailing list discussion about NRD kernels: https://lists.launchpad.net/mimblewimble/msg00636.html

Reply:
Thanks, here's the description I came up https://github.com/marekyggdrasil/docs/blob/master/docs/wiki/transactions/transaction-body.md#nrd-no-recent-duplicate-kernels